_commands/toys/cowsay.md

```diff
@@ -2,23 +2,111 @@
 ---
 
 cowsay
--------
+--
+`cowsay` generates an ASCII picture of a cow saying something provided by  the  user.
 
-ToDo: Needs Completion
 
 ~~~ bash
-*short* example of command usage and output
+$ cowsay Hi There
+ __________ 
+< Hi There >
+ ---------- 
+        \   ^__^
+         \  (oo)\_______
+            (__)\       )\/\
+                ||----w |
+                ||     ||
 ~~~
 
 <!--more-->
 
+
+
 ### Useful Options / Examples
+There are many useful options for this intersting cowsay command.
+The complete list is: cowsay  `[-e  eye_string]` `[-f cowfile]` `[-h]` `[-l]` `[-n]` `[-T tongue_string]` `[-W column]` `[-bdgpstwy]`
+
+`-e` allows the user to customize the cow's eyes which are defaulted to "oo" as seen above.
+`-T` allows the user to add a tongue string to the cow image. By default there is no tongue string.
+`-W` allows the user to customize the number of character printed in the message before the message wraps around
+`-[-bdgpstwy]` are all options to define different pre-made cows emotional states. For example, adding the -g flag will result in a greedy cow to print and -s will result in a stoned cow to print.
 
-#### Example command
+#### cowsay [-bdgpstwy] [`user_input`]
+
+~~~ bash
+$ cowsay -s hi there
+ __________ 
+< hi there >
+ ---------- 
+        \   ^__^
+         \  (**)\_______
+            (__)\       )\/\
+             U  ||----w |
+                ||     ||
+~~~
 
 ##### Break it down
+There are several provided modes which change the appearance of the cow depending on its particular emotional/physical state
+The user can set this state using these flags
+####cowsay [-f cowfile] [`user_string`]
+
+~~~ bash
+$ cowsay -f tux hi there
```

Comment:
![screen shot 2016-03-22 at 10 38 16 pm](https://cloud.githubusercontent.com/assets/7095187/13974071/0ab16a14-f07f-11e5-9df6-d3d0348629c4.png)

Something a little weird on that line. Newlines usually help with this.